tests/integration/conftest.py

```diff
@@ -765,6 +765,78 @@ async def bird(ops_test):
     )
 
 
+@pytest_asyncio.fixture(scope="module")
+async def bird_container_ip(ops_test, bird):
+    bird_app = ops_test.model.applications["bird"]
+    bird_unit = bird_app.units[0]
+
+    cmd = f"run --unit {bird_unit.name} -- sudo sysctl -w net.ipv4.ip_forward=1"
+    rc, stdout, stderr = await ops_test.juju(*shlex.split(cmd))
+    assert rc == 0, f"Failed to enable IP forwarding: {(stdout or stderr).strip()}"
+
+    cmd = f"run --unit {bird_unit.name} -- sudo apt install -y jq"
+    rc, stdout, stderr = await ops_test.juju(*shlex.split(cmd))
+    assert rc == 0, f"Failed to install jq: {(stdout or stderr).strip()}"
+
+    log.info(f"Creating ubuntu container on bird unit {bird_unit.name}")
+    cmd = f"run --unit {bird_unit.name} -- sudo lxd init --auto"
+    rc, stdout, stderr = await ops_test.juju(*shlex.split(cmd))
+    assert rc == 0, f"Failed to initialize lxd: {(stdout or stderr).strip()}"
+
+    cmd = f"run --unit {bird_unit.name} -- sudo lxc launch images:ubuntu/22.04 ubuntu-container"
+    rc, stdout, stderr = await ops_test.juju(*shlex.split(cmd))
+    assert rc == 0, f"Failed to launch ubuntu container: {(stdout or stderr).strip()}"
+
+    cmd = f'run --unit {bird_unit.name} -- sudo lxc list --format=json ubuntu-container | jq -r ".[].state.network.eth0.addresses | .[0].address"'
+    rc, stdout, stderr = await ops_test.juju(*shlex.split(cmd))
+    assert rc == 0, f"Failed to get container IP: {(stdout or stderr).strip()}"
+
+    container_ip = stdout
+    log.info(f"Ubuntu container IP {container_ip}")
+    return container_ip
```

Comment:
Small nit. What do you think about refactor this part into a function? I think a function would be cleaner.

Reply:
which part specifically?

Reply:
As an after thought, I don't think it would add much value. Although the pattern of `cmd + rc,stdout,stderr + assert` is repeated throughout the fixture, refactoring to a function will have the same effect due to the need for `stdout` (e.g. line `795`).